lib/decorators/activity-log.js

```diff
@@ -21,11 +39,32 @@ module.exports = settings => {
 
     return Promise.all(promises)
       .then(logChangedBys => {
-        const activityLog = c.activityLog.map(log => {
+        let activityLog = c.activityLog.filter(filterStatusChanges);
+
+        activityLog = activityLog.reduceRight((store, item, index) => {
+          const itemStatus = getItemStatus(item);
+          const statusChange = getStatusChange(item);
+
+          if (store.length && statusChange && statusChange !== itemStatus) {
+            store[store.length - 1].status = itemStatus;
+            return store;
+          } else {
+            return [
+              ...store,
+              {
+                ...item,
+                status: itemStatus,
+                action: statusChange || itemStatus
+              }
+            ];
+          }
+        }, []);
+
+        activityLog = activityLog.map(log => {
           log.changedBy = logChangedBys.find(profile => profile && profile.id === log.changedBy) || {};
           delete log.event.meta.user.access_token;
           return log;
-        });
+        }).reverse();
```

Comment:
Might be worth making this an explicit `sortBy`, not necessarily because there's any reason to believe it's not correctly sorted, but because it makes it more obvious within the context of this file what the expected sort order is (and less potentially sensitive to breaking as a result of changes elsewhere).